aws-event-bridge/src/main/scala/akka/stream/alpakka/aws/eventbridge/javadsl/EventBridgePublisher.scala

```diff
@@ -0,0 +1,99 @@
+   */
+  def createFlowSeq(
+      settings: EventBridgePublishSettings,
+      eventBridgeClient: EventBridgeAsyncClient
+  ): Flow[PutEventsRequestEntry, PutEventsResponse, NotUsed] =
+    akka.stream.alpakka.aws.eventbridge.scaladsl.EventBridgePublisher.flow(settings)(eventBridgeClient).asJava
+
+  /**
+   * creates a [[akka.stream.javadsl.Flow Flow]] to publish messages to an EventBridge using an [[software.amazon.awssdk.services.eventbridge.EventBridgeAsyncClient EventBridgeAsyncClient]]
+   */
+  def createFlowSeq(
+      eventBridgeClient: EventBridgeAsyncClient
+  ): Flow[Seq[PutEventsRequestEntry], PutEventsResponse, NotUsed] =
+    akka.stream.alpakka.aws.eventbridge.scaladsl.EventBridgePublisher
+      .flowSeq(EventBridgePublishSettings())(eventBridgeClient)
+      .asJava
+
+  /**
+   * creates a [[akka.stream.javadsl.Flow Flow]] to publish messages to an EventBridge using an [[software.amazon.awssdk.services.eventbridge.EventBridgeAsyncClient EventBridgeAsyncClient]]
+   */
+  def createPublishFlow(settings: EventBridgePublishSettings,
+                        eventBridgeClient: EventBridgeAsyncClient): Flow[PutEventsRequest, PutEventsResponse, NotUsed] =
+    akka.stream.alpakka.aws.eventbridge.scaladsl.EventBridgePublisher.publishFlow(settings)(eventBridgeClient).asJava
+
+  /**
+   * creates a [[akka.stream.javadsl.Flow Flow]] to publish a message using an [[software.amazon.awssdk.services.eventbridge.EventBridgeAsyncClient EventBridgeAsyncClient]]
+   */
+  def createPublishFlow(eventBridgeClient: EventBridgeAsyncClient): Flow[PutEventsRequest, PutEventsResponse, NotUsed] =
+    akka.stream.alpakka.aws.eventbridge.scaladsl.EventBridgePublisher
+      .publishFlow(EventBridgePublishSettings())(eventBridgeClient)
+      .asJava
+
+  /**
+   * creates a [[akka.stream.javadsl.Sink Sink]] to publish a message to an EventBridge using an [[software.amazon.awssdk.services.eventbridge.EventBridgeAsyncClient EventBridgeAsyncClient]]
+   */
+  def createSink(eventBridgeClient: EventBridgeAsyncClient): Sink[PutEventsRequestEntry, CompletionStage[Done]] =
+    createFlow(EventBridgePublishSettings(), eventBridgeClient)
+      .toMat(Sink.ignore(), Keep.right[NotUsed, CompletionStage[Done]])
+
+  /**
+   * creates a [[akka.stream.javadsl.Sink Sink]] to publish single message to an EventBridge using an [[software.amazon.awssdk.services.eventbridge.EventBridgeAsyncClient EventBridgeAsyncClient]]
+   */
+  def createSink(settings: EventBridgePublishSettings,
+                 eventBridgeClient: EventBridgeAsyncClient): Sink[PutEventsRequestEntry, CompletionStage[Done]] =
+    createFlow(settings, eventBridgeClient)
+      .toMat(Sink.ignore(), Keep.right[NotUsed, CompletionStage[Done]])
+
+  /**
+   * creates a [[akka.stream.javadsl.Sink Sink]] to publish messages to an EventBridge based on the message topic arn using an [[software.amazon.awssdk.services.eventbridge.EventBridgeAsyncClient EventBridgeAsyncClient]]
```

Comment:
ARN?  https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html ?

Reply:
Yes correct, that's what I meant by `arn` amazon resource name. I will remove that it's wrong.

For SNS it made sense to support publishing to different SNS topics (queues) - different resource in AWS. I was thinking about applying similar approach, even thought most likely there will be only one eventBridge most of the time where a service wants to publish messages. 

It is supported on the PutEventsRequestEntry level.  So one PutEventsRequest - can contain messages to different event buses.  Currently the API is very light in the sense that you have to prepare the the AWS request object yourselves, it would be possible to prepare nicer API (flow, sink), which would take in the `source`, `eventbus`, and potentially the resources and and then the caller would have to provide the detailType and detail. That's the most common usecase in which I'm using it now.

DetailType and detail are both typed as string - type is to describe the definition of what goes into the detail. Serialization comes into mind but then it would limit the possibilities of the routing inside the EventBus.